lib/user.rb

```diff
@@ -0,0 +1,18 @@
+require 'data_mapper'
+
+class User
+  include DataMapper::Resource
+
+  property :id,       Serial
+  property :name,     String,     :required => true
+  property :username, String,     :required => true, :unique => true
+  property :email,    String,     :required => true, :unique => true
+  property :password, BCryptHash, :required => true
+
+  def self.authenticate(username:, password:)
+    user = all(username: username)
```

Comment:
Bit of an ambiguous name, took a while for me to work out that users was an array of all users.

Reply:
fair

Reply:
paul to ask coaches about this...

Reply:
fixed